plugin/clone/include/clone_common.h

```diff
@@ -0,0 +1,33 @@
+/*
+  Copyright (C) 2022-2023 Laurynas Biveinis
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; version 2 of the License.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA */
+
+#ifndef CLONE_COMMON_H
+#define CLONE_COMMON_H
+
+#include "sql/handler.h"
+
+namespace myclone {
+
+class Ha_clone_common_cbk : public Ha_clone_cbk {
+ public:
+  [[nodiscard]] int precopy(THD *thd, uint task_id) override;
```

Comment:
It should be fine to add these directly into `Ha_clone_cbk`. Is there any particular reason to separate it into another class?

Reply:
`Ha_clone_cbk` is an almost interface-like class in the core server. Its all descendant classes, this one included, are implemented in the clone plugin and have substantial code. Folding `Ha_clone_common_cbk` into `Ha_clone_cbk` would result in required definitions being pulled up from plugin to core too, at least.

Reply:
Sounds good then since `Ha_clone_cbk` is in `sql/handler.h` and keeping this code in the plugin is better.